tests/src/JIT/Performance/CodeQuality/HWIntrinsic/X86/PacketTracer/PacketTracer.cs

```diff
@@ -0,0 +1,253 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+using System.Runtime.Intrinsics.X86;
+using static System.Runtime.Intrinsics.X86.Avx;
+using static System.Runtime.Intrinsics.X86.Avx2;
+using static System.Runtime.Intrinsics.X86.Sse2;
+using System.Runtime.Intrinsics;
+using System;
+
+using ColorPacket256 = VectorPacket256;
+
+internal class Packet256Tracer
+{
+    public int Width { get; private set; }
+    public int Hight { get; private set; }
+    private static readonly int MaxDepth = 5;
+
+    private static readonly Vector256<float> SevenToZero = SetVector256(7f, 6f, 5f, 4f, 3f, 2f, 1f, 0f);
+
+    public Packet256Tracer(int _width, int _hight)
+    {
+        if (_width % VectorPacket256.Packet256Size != 0)
+        {
+            _width += VectorPacket256.Packet256Size - _width % VectorPacket256.Packet256Size;
+        }
+        Width = _width;
+        Hight = _hight;
+    }
+
+    internal unsafe void RenderVectorized(Scene scene, int* rgb)
+    {
+        Camera camera = scene.Camera;
+        for (int y = 0; y < Hight; y++)
+        {
+            int stride = y * Width;
+            for (int x = 0; x < Width; x += VectorPacket256.Packet256Size)
+            {
+                float fx = (float)x;
+                Vector256<float> Xs = Add(SetAllVector256(fx), SevenToZero);
+                var dirs = GetPoints(Xs, SetAllVector256<float>(y), camera);
+                var rayPacket256 = new RayPacket256(camera.Pos, dirs);
+                var SoAcolors = TraceRay(rayPacket256, scene, 0);
+ 
+                var AoS = SoAcolors.FastTranspose();
+                var intAoS = AoS.ConvertToIntRGB();
+
+                int* output = &rgb[(x + stride) * 3];
```

Comment:
Without diving in to the rest of the code, the math here for the indexer is "magic".... A comment would be nice

Reply:
Will do.